ui/mirage/config.js

```diff
@@ -335,14 +335,12 @@ export default function() {
     const firstRegion = regions.first();
     return {
       ServerRegion: firstRegion ? firstRegion.id : null,
-      Members: this.serialize(agents.all()),
+      Members: this.serialize(agents.all()).map(({ member }) => ({ ...member })),
     };
   });
 
   this.get('/agent/self', function({ agents }) {
-    return {
-      member: this.serialize(agents.first()),
-    };
+    return agents.first();
```

Comment:
Serialization happens by default. Before this value is returned Mirage is checking is there is a custom Serializer if not it'll apply the behavior in ApplicationSerializer.

We should serialize directly within the route if we're looking to mutate the entity or list of entities because we're working with a collection of ModelClasses that don't have access to methods like `map`, `reduce` and `filter`.

tl;dr - We serialize when we need to use Array and Object Methods before returning out of the route like we should in `agent/members`. The default behavior is to serialize the Collection before returning it and we get that behavior for free because Mirage is already doing this by default.

Reply:
Ah got it, thanks!